airbyte-ci/connectors/pipelines/pipelines/helpers/run_steps.py

```diff
@@ -0,0 +1,233 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+"""The actions package is made to declare reusable pipeline components."""
+
+from __future__ import annotations
+from dataclasses import dataclass, field
+import inspect
+from pipelines import main_logger
+
+from typing import TYPE_CHECKING, Callable, Dict, List, Tuple, Union
+
+import asyncer
+from pipelines.models.steps import StepStatus, StepResult
+
+RESULTS_DICT = Dict[str, StepResult]
+ARGS_TYPE = Union[Dict, Callable[[RESULTS_DICT], Dict]]
+
+if TYPE_CHECKING:
+    from pipelines.models.steps import Step, StepResult
+
+
+@dataclass
+class RunStepOptions:
+    """Options for the run_step function."""
+    fail_fast: bool = True
+    skip_steps: List[str] = field(default_factory=list)
+    log_step_tree: bool = True
+
+
+@dataclass(frozen=True)
+class StepToRun:
+    """
+    A class to wrap a Step with its id and args.
+
+    Used to coordinate the execution of multiple steps inside a pipeline.
+    """
+    id: str
+    step: Step
+    args: ARGS_TYPE = field(default_factory=dict)
+    depends_on: List[str] = field(default_factory=list)
```

Comment:
I would find it more correct and robust if the `depends_on` type was a list of `StepToRun` instances

Reply:
Im not full against that but why is that?

Reply:
A child could know its parents if `depends_on` was a `StepToRun`

Reply:
And it limits the  id typos as we'd pass around a variable